pyquil/paulis.py

```diff
@@ -69,23 +73,22 @@ class PauliTerm(object):
     """A term is a product of Pauli operators operating on different qubits.
     """
 
-    def __init__(self, op, index, coefficient=1.0):
+    def __init__(self, op: str, index: int, coefficient: Union[int, float, complex] = 1):
         """ Create a new Pauli Term with a Pauli operator at a particular index and a leading
         coefficient.
 
-        :param string op: The Pauli operator as a string "X", "Y", "Z", or "I"
-        :param int index: The qubit index that that operator is applied to.
-        :param float coefficient: The coefficient multiplying the operator, e.g. 1.5 * Z_1
+        :param op: The Pauli operator as a string "X", "Y", "Z", or "I"
+        :param index: The qubit index that that operator is applied to.
+        :param coefficient: The coefficient multiplying the operator, e.g. 1.5 * Z_1
         """
         assert op in PAULI_OPS
-        assert _valid_qubit(index)
 
         self._ops = OrderedDict()
         if op != "I":
+            assert _valid_qubit(index)
```

Comment:
Is there a good reason to move this inside the 'if' statement? Do we want to allow things like `PauliTerm(“I”, -1)`?

Reply:
updated it a little bit. I want to say `sI()` sans qubit number

Reply:
It does still strike me as weird that I can do things like
```
>>> from pyquil.paulis import *
>>> print (PauliTerm("I", -1.3434))
(1+0j)*I
>>> print (PauliTerm("X", None))
(1+0j)*XNone
```
I think it's a bad idea if we allowed such things.

Reply:
good catch. fixed in 6a8457ce6d93dd63057cbfa47993e4d76988ad7a